src/MoBi.Presentation/Presenter/HierarchicalSpatialStructurePresenter.cs

```diff
@@ -100,21 +100,18 @@ public void Handle(AddedEvent eventToHandle)
          if (entity.IsAnImplementationOf<IDistributedParameter>())
             return;
 
+         //not in the spatial structure? nothing to do here in the first place
+         if (!entityIsInSpatialStructure(entity))
+            return;
 
          var dto = _objectBaseMapper.MapFrom(entity);
 
          // A root container in a spatial structure should be added to the root of the tree view
          // The root container will have ParentContainer == null
-         if (entityIsInSpatialStructure(entity) && !entityIsRootContainer(entity))
-            _view.Add(dto, _objectBaseMapper.MapFrom(entity.ParentContainer));
-         else
-         {
-            //the object that has changed does not belong into the spatial structure. Nothing do to
-            if (!Equals(eventToHandle.Parent, _spatialStructure))
```

Comment:
This was wrong here. When addnga neighborhood, the parent is NOT the spatial structure but the NeighborhoodsContainer